mono/metadata/icall.c

```diff
@@ -6596,6 +6596,12 @@ ves_icall_System_Buffer_SetByteInternal (MonoArray *array, gint32 idx, gint8 val
 	mono_array_set (array, gint8, idx, value);
 }
 
+ICALL_EXPORT void
+ves_icall_System_Buffer_MemcpyInternal (gpointer dest, gpointer src, gint32 count)
```

Comment:
Last parameter should be gsize or gssize or gint64 or guint64.
src should be gconstpointer.

Reply:
For safety, maybe call memmove instead of memcpy. I thought  caller might check for overlap though.